src/guide/deployment/mongodb.md

```diff
@@ -0,0 +1,155 @@
+
+    ${mcfg.extraConfig}
+  '';
+in {
+  environment.systemPackages = [
+    pkgs.mongodb-tools
+  ];
+
+  services.mongodb = {
+    enable = true;
+    dbpath = "/srv/mongodb";
+    bind_ip = fclib.mkPlatform (lib.concatStringsSep "," listenAddresses);
+    pidFile = "/run/mongodb.pid";
+    package = throw "Please replace this `throw ...` expression with 'builtins.storePath /nix/store/…'";
+    enableAuth = true;
+
+    # WARNING: this will only work on the very first install.
+    # You need to change MongoDB's root password manually later on.
+    initialRootPassword = throw "Please replace this `throw ...` expression with the initial root password of the mongodb root account";
+
+    # WARNING: don't set this on the initial install of MongoDB!
+    # Otherwise the initialRootPassword won't be set.
+    #replSetName = "replica";
+  };
+
+  systemd.services.mongodb = {
+    preStart = "echo never > /sys/kernel/mm/transparent_hugepage/defrag";
+    postStop = "echo always > /sys/kernel/mm/transparent_hugepage/defrag";
+    # intial creating of journal takes ages
+    serviceConfig.TimeoutStartSec = fclib.mkPlatform 1200;
+    serviceConfig.LimitNOFILE = 64000;
+    serviceConfig.LimitNPROC = 32000;
+    serviceConfig.Restart = "always";
+    serviceConfig.ExecStart = lib.mkForce ''
+      ${mcfg.package}/bin/mongod --config ${mongoCnf}
+    '';
+    reload = ''
+      if [[ -f ${mcfg.pidFile} ]]; then
+        kill -USR1 $(< ${mcfg.pidFile} )
+      fi
+    '';
+  };
+
+  users.users.mongodb = {
+    shell = "/run/current-system/sw/bin/bash";
+    home = "/srv/mongodb";
+  };
+
+  flyingcircus.infrastructure.preferNoneSchedulerOnSsd = true;
+
+  flyingcircus.activationScripts = {
+    mongodb-dirs = lib.stringAfter [ "users" "groups" ] ''
+      install -d -o mongodb /{srv,var/log}/mongodb
+    '';
+  };
+
+  security.sudo.extraRules = [
+    # Service users may switch to the mongodb system user
+    {
+      commands = [ { command = "ALL"; options = [ "NOPASSWD" ]; } ];
+      groups = [ "sudo-srv" "service" ];
+      runAs = "mongodb";
+    }
+    {
+      commands = [ { command = checkMongoCmd; options = [ "NOPASSWD" ]; } ];
+      users = [ "sensuclient" ];
+      runAs = "mongodb";
+    }
+  ];
+
+  services.logrotate.settings = {
+    "/var/log/mongodb/*.log" = {
+      nocreate = true;
+      postrotate = "systemctl reload mongodb";
+    };
+  };
+
+  flyingcircus.services = {
+    sensu-client.checks = {
+      mongodb = {
+        notification = "MongoDB alive";
+        command = ''
+          /run/wrappers/bin/sudo -u mongodb -- ${checkMongoCmd} -d mongodb ${extraCheckArgs}
+        '';
+      };
+      mongodb_feature_compat_version = {
+        notification = "MongoDB is running on an outdated feature compatibility version";
+        command = ''
+          /run/wrappers/bin/sudo -u mongodb -- ${checkMongoCmd} -d mongodb -A feature_compat_version ${extraCheckArgs}
+        '';
+        interval = 600;
+      };
+    };
```

Comment:
Does it make sense to leave this in?

Reply:
yeah, why not. Being documentation I wonder if we should discuss those things in a bit more detail.